cobalt/browser/cobalt_browser_main_parts.h

```diff
@@ -52,7 +62,35 @@ class CobaltBrowserMainParts : public content::ShellBrowserMainParts {
 #endif  // BUILDFLAG(IS_LINUX)
 
  private:
+  // Initializes, but doesn't start, UMA.
+  void SetupMetrics();
+
+  // Starts metrics recording.
+  void StartMetricsRecording();
+
+  // Fetch and, if necessary, initializes MetricsService instance owned
+  // by this class.
+  metrics::MetricsService* GetMetricsService();
+
+  // Fetch and, if necessary, initializes MetricsServicesManager owned by this
+  // class instance.
+  metrics_services_manager::MetricsServicesManager* GetMetricsServicesManager();
+
+  PrefService* local_state();
+
+  base::FilePath cache_directory_;
+
   std::unique_ptr<CobaltMetricsServiceClient> metrics_;
+
+  // Must be destroyed before |local_state_|.
+  std::unique_ptr<metrics_services_manager::MetricsServicesManager>
+      metrics_services_manager_;
+
+  // |metrics_services_manager_| owns this.
+  raw_ptr<CobaltMetricsServicesManagerClient> metrics_services_manager_client_ =
```

Comment:
Why do we need to maintain a reference to this?

Reply:
Good catch. We don't. I was following Chrome's pattern here and they use this reference, but we don't, yet.